ironic/resource_ironic_node_v1.go

```diff
@@ -585,3 +614,140 @@ func changePowerState(client *gophercloud.ServiceClient, d *schema.ResourceData,
 
 	return nil
 }
+
+// setRAIDConfig calls ironic's API to send request to change a Node's RAID config.
+func setRAIDConfig(client *gophercloud.ServiceClient, d *schema.ResourceData) (err error) {
+	var logicalDisks []nodes.LogicalDisk
+	var raid *metal3v1alpha1.RAIDConfig
+
+	raidConfig := d.Get("raid_config").(string)
+	if raidConfig == "" {
+		return nil
+	}
+
+	err = json.Unmarshal([]byte(raidConfig), &raid)
+	if err != nil {
+		return
+	}
+
+	err = checkRAIDConfigure(d.Get("raid_interface").(string), raid)
+	if err != nil {
+		return
+	}
+
+	// Build target for RAID configuration steps
+	logicalDisks, err = ironic.BuildTargetRAIDCfg(raid)
+	if len(logicalDisks) == 0 || err != nil {
+		return
+	}
+
+	// Set root volume
+	if len(d.Get("root_device").(map[string]interface{})) == 0 {
+		logicalDisks[0].IsRootVolume = new(bool)
+		*logicalDisks[0].IsRootVolume = true
+	} else {
+		log.Printf("rootDeviceHints is used, the first volume of raid will not be set to root")
+	}
+
+	// Set target for RAID configuration steps
+	return nodes.SetRAIDConfig(
+		client,
+		d.Id(),
+		nodes.RAIDConfigOpts{LogicalDisks: logicalDisks},
+	).ExtractErr()
+}
+
+// buildBIOSSettings builds bios settings for BIOS clean steps
+func buildBIOSSettings(d *schema.ResourceData, firmwareConfig *bmc.FirmwareConfig) (settings []map[string]string, err error) {
+	driver := d.Get("driver").(string)
+	driverInfo := d.Get("driver_info").(map[string]interface{})
+	driverAddress := strings.Join([]string{driver, "address"}, "_")
```

Comment:
There is at least one driver, for which this procedure does not work: `idrac` (uses both redfish and drac prefixes). Maybe update the bmc library with a function that does not require an address?

Reply:
Yes, you are right, this is a problem for dealing with BIOS settings.
I have 2 ideas:
1. process BIOS setting on installer side, then transfer the result of [BuildBIOSSettings](https://github.com/metal3-io/baremetal-operator/blob/master/pkg/hardwareutils/bmc/access.go#L86) to TF.
2. transfer [BMCDetails](https://github.com/metal3-io/baremetal-operator/blob/master/apis/metal3.io/v1alpha1/baremetalhost_types.go#L217) to TF from installer(thus the address should be correct)

What do you think?

> Maybe update the bmc library with a function that does not require an address?

I am not quite understand your suggestion. Do you mean an improvement on BMO?
This may be a fundamental solution, but we hope that the BIOS&RAID feature can catch up with OCP4.10, so can we consider this after that?

Reply:
Regarding above two ideas, IMO, the 1st idea may be not a standard way, and the installer should not perform such processing. 

I [transfer BMC Address](https://github.com/openshift/installer/pull/5196/commits/7c735d78c1ae8e132ecb1a36b4fadc55156c479d) from installer to here, thus the NewAccessDetails can work properly for all drivers(bmcs). PTAL.

Reply:
IINW, what @dtantsur means is adding a function in here https://github.com/metal3-io/baremetal-operator/blob/4950ec837c58176637bb4f95eca8f629301e43d9/pkg/hardwareutils/bmc/access.go#L87 instead using above function.

Thanks to that, above calculation will work per driver by calling their own function in bmc.

Reply:
@ardaguclu @dtantsur what's the consensus here - should we go ahead with the current approach then fix the idrac issue (with associated new bmc module change in BMO) as a followup?

Reply:
> There is at least one driver, for which this procedure does not work: `idrac` (uses both redfish and drac prefixes). Maybe update the bmc library with a function that does not require an address?

If this is not a explicit bug and just a clean up, we can move forward and merge that PR. What do you think @dtantsur ?

Reply:
Yeah, but let's plan to fix it as soon as possible - we don't want to leave broken code around for too long.

Reply:
Ok perhaps someone can raise a bz to capture the details, then we can take care of that as a follow-up